Documentation/compatibility/ServiceBase-doesnt-propagate-OnStart-exceptions.md

```diff
@@ -0,0 +1,26 @@
+## ServiceBase doesn't propagate OnStart exceptions
+
+### Scope
+
+Minor
+
+### Version Introduced
+
+4.7.1
+
+### Source Analyzer Status
+
+NotPlanned
+
+### Change Description
+
+Exceptions from services that fail to start will now be propagated from ServiceBase.Run
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+On service start, if there is an exception, that exception will be propagated. This should help diagnose cases where services fail to start.
```

Comment:
Add the following:

If this behavior is undesirable, you can opt out of it by adding the following [`<AppContextSwitchOverrides>`](~/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element.md) element to the `<runtime>` section of your application configuration file:

```xml
<AppContextSwitchOverrides value="Switch.System.ServiceProcess.DontThrowExceptionsOnStart=true" />
```

Also, can you opt into the behavior for applications that target previous versions of the .NET Framework?

Reply:
Done